### DISCUSSION TITLE
btrfs and filesystem traversal

### DISCUSSION THREAD

**Discussion by iameru** [CyberShadow/aconfmgr#171] [Q&A] [answered]
I was trying aconfmgr but stopped. I have a filesystem with lots of btrfs subvolumes mounted and lots of snapshots.
aconfmgr tries to walk through all of them. A hot fix probably would be to check for btrfs and their mountpoint and only take `/` and `/home/` and other relevant ones into consideration.
For now this takes unusably long to walk through.

Not sure if bug or issue, decide for yourself.

On a side note, I've been expecting aconfmgr to check `$PATH` and only look there or check the installed packages of the system, not to traverse through everything, everywhere. Why did you chose this path?

**Comment by CyberShadow** [comment, marked as answer]
Snapshots are state, so you have to add ignore rules for them.

> I have a filesystem with lots of btrfs subvolumes mounted and lots of snapshots.

This is an unusual setup.

> A hot fix probably would be to check for btrfs and their mountpoint and only take / and /home/ and other relevant ones into consideration.

What is relevant or not depends entirely on your setup and goal.

> For now this takes unusably long to walk through.

Ignore rules should help here. If your snapshots follow a consistent naming scheme, one ignore rule might be enough (e.g. `*/@*-????-??-??`).

Note that you can also generate the ignore rule list programmatically, e.g. by enumerating your subvolumes in your aconfmgr configuration.

> Not sure if bug or issue, decide for yourself.

I do not see a bug here.

> On a side note, I've been expecting aconfmgr to check $PATH and only look there or check the installed packages of the system, not to traverse through everything, everywhere. Why did you chose this path?

As described in the documentation, aconfmgr aims to describe the state of the entire system, barring some explicit exclusions, which are left up to the user to choose.

`PATH` is not really relevant here. State is any file on the filesystem, and aconfmgr neither assumes judgement of what state a user might consider worth tracking, nor could possibly know whether a file can affect a system's behavior within any definition of what behavior the user might consider worth tracking.
